nft-drop/src/components/Claim/NFTFrame.tsx

```diff
@@ -0,0 +1,121 @@
+import { useWeb3React } from "@web3-react/core";
+import React, { useEffect, useMemo } from "react";
+import styled, { keyframes } from "styled-components";
+
+import { PrimaryText } from "shared/lib/designSystem";
+import {
+  getLogoColorFromColorway,
+  getThemeColorFromColorway,
+} from "../../utils/colors";
+import Logo from "shared/lib/assets/icons/logo";
+import { useNFTDropData } from "../../hooks/nftDataContext";
+import { useNFTDropGlobalState } from "../../store/store";
+
+const Frame = styled.div<{ width: number; height: number }>`
+  display: flex;
+  flex-direction: column;
+  height: ${(props) => props.height}px;
+  width: ${(props) => props.width}px;
+  background: rgba(255, 255, 255, 0.02);
+  backdrop-filter: blur(40px);
+  overflow: hidden;
+`;
+
+const livelyAnimation = (position: "top" | "bottom") => keyframes`
+  0% {
+    background-position-x: ${position === "top" ? 0 : 100}%;
+  }
+
+  50% {
+    background-position-x: ${position === "top" ? 100 : 0}%; 
+  }
+
+  100% {
+    background-position-x: ${position === "top" ? 0 : 100}%;
+  }
+`;
+
+const FrameBar = styled.div<{
+  color: string;
+  position: "top" | "bottom";
+  height: number;
+}>`
+  width: 100%;
+  height: ${(props) => props.height}px;
+  background: ${(props) => `linear-gradient(
+    270deg,
+    ${props.color}00 0%,
+    ${props.color} 50%,
+    ${props.color}00 100%
+  )`};
+  background-size: 200%;
+  animation: 10s ${(props) => livelyAnimation(props.position)} linear infinite;
+
+  &:before {
+    content: "";
+    z-index: -1;
+    position: absolute;
+    ${(props) => props.position}: 0;
+    right: 0;
+    left: 0;
+    background: inherit;
+    filter: blur(${(props) => props.height * 4}px);
+    opacity: 1;
+    transition: opacity 0.3s;
+    height: ${(props) => props.height * 2}px;
+  }
+`;
+
+interface NFTFrameProps {
+  height: number;
+  width: number;
+}
+
+const NFTFrame: React.FC<NFTFrameProps> = ({ height, width }) => {
+  const { active } = useWeb3React();
+  const nftDropData = useNFTDropData();
+  const [, setClaimButtonWidth] = useNFTDropGlobalState("claimButtonWidth");
+
+  useEffect(() => {
+    setClaimButtonWidth(width);
+  }, [setClaimButtonWidth, width]);
+
+  const content = useMemo(() => {
+    if (!active) {
+      return (
+        <PrimaryText fontSize={14} lineHeight={20}>
+          Please connect your wallet
+        </PrimaryText>
+      );
+    }
+
+    if (nftDropData.colorway !== undefined) {
```

Comment:
We need to handle the case where `nftDropData.colorway === undefined`, with a text saying something like "Sorry, you are not qualified for the drop"

Reply:
Okay I see this addressed at the bottom